lusidtools/cocoon/instruments.py

```diff
@@ -335,50 +347,29 @@ def get_unique_identifiers(api_factory: lusid.utilities.ApiClientFactory):
                 continue
             elif len(single_search[0].external_instruments) > 0:
                 name = single_search[0].external_instruments[0].name
+                figi = single_search[0].external_instruments[0].identifiers["Figi"].value
                 break
         names.append(name)
+        figis.append(figi)
 
     enriched_column_name = "LUSID.Name.Enriched"
+    enriched_column_figi = "LUSID.Figi.Enriched"
 
     data_frame[enriched_column_name] = names
+    data_frame[enriched_column_figi] = figis
 
     # Missing mapping for name altogether
     if "name" not in list(mapping_required.keys()):
         mapping_required["name"] = enriched_column_name
+    else:
+        enrich_existing_column(data_frame, mapping_required, "name", enriched_column_name, constant_prefix)
+
+    if "Figi" not in list(instrument_identifier_mapping.keys()):
+        instrument_identifier_mapping["Figi"] = enriched_column_figi
+    else:
+        enrich_existing_column(data_frame, instrument_identifier_mapping, "Figi", enriched_column_figi, constant_prefix)
```

Comment:
I believe someone could also provide Instrument/default/Figi in the mapping. You should call prepare_key (which is also in the instruments file) on the instrument identifier mapping keys first then check for the existence of the full key.

Reply:
Check for figi key [long or short] then use that as the column header to enrich